.github/workflows/unit_test.yml

```diff
@@ -40,3 +40,4 @@ jobs:
       run: coveralls
       env:
         COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} #Github Tokens
```

Comment:
This is not the correct place to have a GitHub token. You will have to provide me with the token and I will add it as a secret on GitHub.

Reply:
Which token are we trying to implement here?

Reply:
My bad , I didn't noticed that github tokens got exposed.

Reply:
the github token is automatically set and it is required because without it the upload coverage to coverall is failing